app/lib/menu.rb

```diff
@@ -17,7 +17,7 @@ def scope(name, svg, children)
     @items[name] = { svg: "#{svg}.svg", children: children }
   end
 
-  def item(name:, controller: name, children: [], visible: true)
```

Comment:
@Ridhwana I made some changes here and would love to know if this is a right approach.

Reason for these changes is I wanted to always have a clear indication re what nav item is the "parent" one so we can highlight the right item without hacking too much in the view.

Reply:
Thanks Pawel, here's my thoughts:

When we define the structure, we already have a hierarchy and a "children" key. This shows who the parent is based on the data structure:
```
{:people=>{:svg=>"group-2-line.svg", :children=>[{:name=>"people", :controller=>"users", :children=>[], :visible=>true}]},
 :content_manager=>
  {:svg=>"dashboard-line.svg",
   :children=>
    [{:name=>"posts", :controller=>"articles", :children=>[], :visible=>true},
     {:name=>"comments", :controller=>"comments", :children=>[], :visible=>true},
     {:name=>"badges",
      :controller=>"badges",
      :children=>[{:name=>"library", :controller=>"badges", :children=>[], :visible=>true}, {:name=>"achievements", :controller=>"badge_achievements", :children=>[], :visible=>true}],
      :visible=>true},
     {:name=>"organizations", :controller=>"organizations", :children=>[], :visible=>true},
     {:name=>"podcasts", :controller=>"podcasts", :children=>[], :visible=>true},
     {:name=>"tags", :controller=>"tags", :children=>[], :visible=>true}]},
 :customization=>
  {:svg=>"tools-line.svg",
   :children=>
    [{:name=>"config", :controller=>"config", :children=>[], :visible=>true},
     {:name=>"html variants", :controller=>"html_variants", :children=>[], :visible=>true},
     {:name=>"display ads", :controller=>"display_ads", :children=>[], :visible=>true},
     {:name=>"navigation links", :controller=>"navigation_links", :children=>[], :visible=>true},
     {:name=>"pages", :controller=>"pages", :children=>[], :visible=>true},
     {:name=>"profile fields", :controller=>"profile_fields", :children=>[], :visible=>false}]},
 :admin_team=>{:svg=>"user-line.svg", :children=>[{:name=>"admin team", :controller=>"permissions", :children=>[], :visible=>true}]},
 :moderation=>
  {:svg=>"mod.svg",
   :children=>
    [{:name=>"reports", :controller=>"reports", :children=>[], :visible=>true},
     {:name=>"mods", :controller=>"mods", :children=>[], :visible=>true},
     {:name=>"moderator actions ads", :controller=>"moderator_actions", :children=>[], :visible=>true},
     {:name=>"privileged reactions", :controller=>"privileged_reactions", :children=>[], :visible=>true}]},
 :advanced=>
  {:svg=>"flashlight-line.svg",
   :children=>
    [{:name=>"broadcasts", :controller=>"broadcasts", :children=>[], :visible=>true},
     {:name=>"response templates", :controller=>"response_templates", :children=>[], :visible=>true},
     {:name=>"sponsorships", :controller=>"sponsorships", :children=>[], :visible=>true},
     {:name=>"developer tools",
      :controller=>"tools",
      :children=>
       [{:name=>"tools", :controller=>"tools", :children=>[], :visible=>true},
        {:name=>"vault secrets", :controller=>"secrets", :children=>[], :visible=>true},
        {:name=>"data update scripts", :controller=>"data_update_scripts", :children=>[], :visible=>false}],
      :visible=>true}]},
 :apps=>
  {:svg=>"palette-line.svg",
   :children=>
    [{:name=>"consumer apps", :controller=>"consumer_apps", :children=>[], :visible=>true},
     {:name=>"listings", :controller=>"listings", :children=>[], :visible=>true},
     {:name=>"welcome", :controller=>"welcome", :children=>[], :visible=>true}]}}
```
Hence, to me this feels redundant and an overhead where we need to keep track of the parent when it's clear in the data structure.

Reply:
👍 I removed redundant data.